### PR TITLE
Nwb extractors

### PR DESCRIPTION
**Pull request by alejoe91**

### CODE REVIEW

File: spikeinterface/core/basesorting.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ class BaseSorting(BaseExtractor):
     Abstract class representing several segment several units and relative spiketrains.
     """
 
-    def __init__(self, sampling_frequency: float, unit_ids: List[int]):
+    def __init__(self, sampling_frequency: float, unit_ids: List):
```
Comment (samuelgarcia) [inline spikeinterface/core/basesorting.py:13]: In fact it is not a list but also array are possible.

File: spikeinterface/extractors/nwbextractors.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
+import uuid
```
Comment (samuelgarcia) [inline spikeinterface/extractors/nwbextractors.py:1]: Do we need uuid for reading ?

File: spikeinterface/extractors/nwbextractors.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
+        with NWBHDF5IO(self._path, 'r') as io:
+            nwbfile = io.read()
+            es = nwbfile.acquisition[self._electrical_series_name]
+            es_channel_ids = np.array(es.electrodes.table.id[:])[es.electrodes.data[:]].tolist()
+            channel_inds = [es_channel_ids.index(id) for id in channel_indices]
+            if np.array(channel_indices).size > 1 and np.any(np.diff(channel_indices) < 0):
```
Comment (samuelgarcia) [inline spikeinterface/extractors/nwbextractors.py:178]: channel_indices can be a slice!. Is it handled ?

File: spikeinterface/extractors/nwbextractors.py [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+
+
+class NwbSortingExtractor(BaseSorting):
+    extractor_name = 'NwbSorting'
+    installed = HAVE_NWB  # check at class level if installed or not
+    is_writable = True
+    mode = 'file'
+    installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+
+    def __init__(self, file_path, electrical_series: str = None, sampling_frequency: float = None):
```
Comment (samuelgarcia) [inline spikeinterface/extractors/nwbextractors.py:193]: For consistency I would put electrical_series_name here.
It will help for the `read_nwb()`

**Comment by samuelgarcia** [comment]
@alejoe91 : I made a comment and also I push a commit.
Check it.

**Comment by samuelgarcia** [comment]
@alejoe91 : see this before merging https://gin.g-node.org/NeuralEnsemble/ephy_testing_data/issues/55#issuecomment-1486

**Comment by alejoe91** [comment]
@samuelgarcia can we merge this?

**Comment by samuelgarcia** [comment]
I made some comments can you check ?

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**